### PR TITLE
Update nnnn-async-await.md

### PR DESCRIPTION
**Pull request by benrimmington**
* ~~Add table of contents~~.
* Fix errors and typos.
* ~~Update forums link~~.

### CODE REVIEW

**Comment by ktoso** [comment]
Hi there, thanks for the typo fixes though we're not so sure about the ToC -- it gets hard to maintain manually.

**Review by ktoso — CHANGES_REQUESTED** [review]
Could you remove the ToC and we can get in just the links and typos?

File: proposals/nnnn-async-await.md [hunk 1]
```diff
@@ -214,11 +214,11 @@ When control returns to an asynchronous function, it picks up exactly where it w
...
...larly important because *suspensions interrupt atomicity*.  For example, if an asynchronous function is running within a given context that is protected by a serial queue, reaching a suspension point means that other code can be interleaved on that same serial queue.  A classic but somewhat hackneyed example where this atomicity matters is modeling a bank: if a deposit is credited to one account, but the operation suspends before processing a matched withdrawal, it creates a window where those funds can be double-spent.  A more germane example for many Swift programmers is a UI thread: the suspension points are the points where the UI can be shown to the user, so programs that build part of their UI and then suspend risk presenting a flickering, partially-constructed UI.  (Note that suspension points are also called out explicitly in code using explicit callbacks: the suspension happens between the point where the outer function returns and the callback starts running.)  Requiring that all suspension points are marked allows programmers to safely assume that places without suspension points will behave atomically, as well as to more easily recognize problematic non-atomic patterns.
```
Comment (benrimmington) [inline proposals/nnnn-async-await.md:217]: (Removed `a` before `modeling a bank`.)

File: proposals/nnnn-async-await.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ This proposal defines the semantics of asynchronous functions. However, it does
 This proposal draws some inspiration (and most of the Motivation section) from an earlier proposal written by 
 [Chris Lattner](https://github.com/lattner) and [Joe Groff](https://github.com/jckarter), available [here](https://gist.github.com/lattner/429b9070918248274f25b714dcfc7619). That proposal itself is derived from a proposal written by [Oleg Andreev](https://github.com/oleganza), available [here](https://gist.github.com/oleganza/7342ed829bddd86f740a). It has been significantly rewritten (again), and many details have changed, but the core ideas of asynchronous functions have remained the same.
 
-Swift-evolution thread: [Discussion thread topic for that proposal](https://forums.swift.org/t/concurrency-asynchronous-functions/41619)
+Swift-evolution thread: [\[Concurrency\] Asynchronous functions](https://forums.swift.org/t/concurrency-asynchronous-functions/41619)
```
Comment (ktoso) [inline proposals/nnnn-async-await.md]:
```suggestion
Swift-evolution thread: [Discussion thread topic for that proposal](https://forums.swift.org/t/concurrency-asynchronous-functions/41619)
```

to be consistent with all other proposal documents

File: proposals/nnnn-async-await.md [hunk 3]
```diff
@@ -82,7 +82,7 @@ func processImageData2(completionBlock: (result: Image?, error: Error?) -> Void)
 
 processImageData2 { image, error in
     guard let image = image else {
-        error("No image today")
+        display("No image today", error)
```
Comment (ktoso) [inline proposals/nnnn-async-await.md:85]: ok makes sense; error is just `Error` so would not make sense to apply it.

File: proposals/nnnn-async-await.md [hunk 4]
```diff
@@ -92,7 +92,7 @@ processImageData2 { image, error in
 The addition of [`Result`](https://github.com/apple/swift-evolution/blob/main/proposals/0235-add-result.md) to the standard library improved on error handling for Swift APIs. Asynchronous APIs were one of the [main motivators](https://github.com/apple/swift-evolution/blob/main/proposals/0235-add-result.md#asynchronous-apis) for `Result`: 
 
 ```swift
-func processImageData2(completionBlock: (Result<Image>) -> Void) {
+func processImageData2(completionBlock: (Result<Image, Error>) -> Void) {
```
Comment (ktoso) [inline proposals/nnnn-async-await.md:95]: good, thanks

File: proposals/nnnn-async-await.md [hunk 5]
```diff
@@ -127,8 +127,8 @@ It's easier to properly thread the error through when using `Result`, making the
 Conditionally executing an asynchronous function is a huge pain. For example, suppose we need to "swizzle" an image after obtaining it. But, we sometimes have to make an asynchronous call to decode the image before we can swizzle. Perhaps the best approach to structuring this function is to write the swizzling code in a helper "continuation" closure that is conditionally captured in a completion handler, like this:
 
 ```swift
-func processImageData3(recipient: Person, completionBlock: (result: Image) -> Void) {
-    let swizzle: (contents: image) -> Void = {
+func processImageData3(recipient: Person, completionBlock: (_ result: Image) -> Void) {
+    let swizzle: (_ contents: Image) -> Void = {
```
Comment (ktoso) [inline proposals/nnnn-async-await.md:131]: thanks

File: proposals/nnnn-async-await.md (hunk 1 above)
Comment (ktoso) [reply proposals/nnnn-async-await.md:217]: Thanks for commenting on the change, good

**Review by ktoso — APPROVED** [review]
Thanks, this looks good; makes the snippets more compile-able

File: proposals/nnnn-async-await.md (hunk 2 above)
Comment (benrimmington) [reply proposals/nnnn-async-await.md]: @ktoso "Discussion thread topic for that proposal" is from the [template](https://github.com/apple/swift-evolution/blob/main/proposal-templates/0000-swift-template.md), but most of the accepted proposals have changed it to the actual name of their forums topic.

Comment (ktoso) [reply proposals/nnnn-async-await.md]: I see, let's fix it in all proposals then :) 

also https://github.com/DougGregor/swift-evolution/blob/structured-concurrency/proposals/nnnn-structured-concurrency.md and the actors one.

do you want to send in the PRs?

Comment (benrimmington) [reply proposals/nnnn-async-await.md]: > do you want to send in the PRs?

Yes, I'll include the changes in my other PRs.

**Pull request merged**